lib/decimal.ex

```diff
@@ -1510,6 +1510,24 @@ defmodule Decimal do
     end
   end
 
+  @doc """
+  Returns the sum of all elements in the given enumerable.
+
+  ## Examples
+
+      iex> Decimal.sum([1, "2", Decimal.new("3")])
+      #Decimal<6>
+
+      iex> Decimal.sum([])
+      #Decimal<0>
+
+  """
+  doc_since("1.9.0")
+  @spec sum(Enumerable.t()) :: t()
+  def sum(enumerable) do
+    Enum.reduce(enumerable, Decimal.new(0), &Decimal.add/2)
```

Comment:
Can't this be implemented with `Enum.reduce/2`? If so, my two cents are to not add this to the API and have people write

```elixir
Enum.reduce(enumerable, &Decimal.add/2)
```

Reply:
Regarding `Enum.reduce/2` vs `Enum.reduce/3`, good question, the idea was to mimic `Enum.sum` which doesn't crash on empty enumerable.

I tend to agree that we shouldn't introduce API if we can easily achieve the same outcome with existing functions, but, and this is very subjective, I feel like in this particular case it's such a common operation to perform that skipping the reduce step helps readability even if just a little bit.

Reply:
I'm not an avid user of Decimal (never had the chance really) so I'll let others decide how common this is :)